.github/workflows/docker_release.yml

```
@@ -0,0 +1,74 @@
+name: Release Docker images
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - massi/docker
+    tags:
+      - v*
+
+env:
+  DOCKER_REPO_NAME: deepset/haystack
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USER }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: $DOCKER_REPO_NAME
+
+      - name: Build base images
+        uses: docker/bake-action@v2
+        env:
+          IMAGE_TAG_SUFFIX: ${{ steps.meta.outputs.version }}
+        with:
+          workdir: docker
+          targets: base
+          push: true
+
+      - name: Build Haystack images
```

Just for consistency with the above

```suggestion
      - name: Build api images
```